File: pinyougou-manager-web/src/main/webapp/js/service/specificationService.js
/**
 * 规格信息的服务层
 */
app.service("specificationService",function($http){

    //全查-分页-模糊
    this.findPage = function(searchEntity,page,rows){
        return $http.post("../spec/findPage.do?page="+page+"&rows="+rows,searchEntity);
    },
    //批量删除
    this.deleteSpecification = function(ids){
        return $http.post("../spec/deleteSpecification.do?ids="+ids);
    },
    //保存信息
    this.saveSpecification = function (entity) {
        return $http.post("../spec/saveSpecification.do",entity);
    },
    //修改回显
    this.findOne = function(id){
        return $http.get("../spec/findOne.do?id="+id);
    },
    //全查规格信息，模板管理select2插件回显时用的
    this.selectSpecOptionList = function () {
        return $http.get("../spec/selectSpecOptionList.do");
    }

});

File: pinyougou-sellergoods-interface/src/main/java/com/pinyougou/sellergoods/service/SpecificationService.java
package com.pinyougou.sellergoods.service;

import com.pinyougou.pojo.TbSpecification;
import com.pinyougou.pojogroup.Specification;
import entity.PageResult;
import entity.Result;

import java.util.List;
import java.util.Map;

/**
 * 规格信息业务逻辑层接口
 */
public interface SpecificationService {
    //全查-分页-模糊
    PageResult findPage(TbSpecification specification, int page, int rows);

    //批量删除，规格信息以及规格选项信息
    void deleteSpecification(Long[] ids);

    //保存规格信息以及规格选项信息
    void saveSpecification(Specification specification);

    //根据规格id查询出规格信息以及规格选项信息
    Specification findOne(Long id);

    //全查规格信息，模板管理select2插件回显时用的
    List<Map> selectSpecOptionList();
}
